jdaviz/app.py

```diff
@@ -381,9 +381,21 @@ def _on_layers_changed(self, msg):
         else:
             raise NotImplementedError(f"cannot recognize new layer from {msg}")
 
+        def _new_layer_name(index):
```

Comment:
If you take this function out from all the nesting, it would be possible to test.